### ISSUE TITLE
False Negatives

### ISSUE THREAD

**Issue by sivak95** [Daniel595/Jetson_nano_face_recognition#2]
Hello, I was able to get it to work with video on the jetson nano with a MIPI-CSI camera and I'm getting 25-30 fps with just one face in the frame. Amazing work with TensorRT and the code optimizations like shared memory utilization! However, if I don't look directly at the camera (even slightly above or below), my face is classified as unknown and I have currently trained the SVM with 5 images of myself. I have made an implementation of dlib with knn classification where I was getting almost zero false positives and negatives with a single training image. I was wondering what I can do to fix it in this case ( More training images, hyperparameter tuning, adjusting confidence threshold ) ? Also, apart from using TensorRT, what other differences are present in this cpp implementation of MTCNN? Is there a noticeable accuracy difference with the original implementation?

**Comment by Daniel595** [comment]
Hello sivak, I'm very glad that you got it run on your nano :)

Concerning the SVM accuracy I faced the same problem multiple times. Once I changed the number of trained classes or the number of used samples per class I had to do fine-tuning manually.  

My first shot would be to have a look at the face_classifiers [prediction method](https://github.com/Daniel595/Jetson_nano_face_recognition/blob/bcb352a77a12045979f294428b2cf233ad33f3eb/src/face_classifier.cpp#L129-L194 ) where a voting mechanism is used. Try to play around with 'sum_threshhold' 'mean_threshhold'([this block](https://github.com/Daniel595/Jetson_nano_face_recognition/blob/bcb352a77a12045979f294428b2cf233ad33f3eb/src/face_classifier.cpp#L168-L174)). Currently, while predicting the method should produce some outputs on your terminal which shows you the votings for the single classes. With those messages I kind of debugged this and tried to estimate a good threshhold. BTW - the fact that those debug messages are still active shows that it's not satisfactory for me and I'm not surprised getting an issue like that. 

Currently I don't have the system setup running and I can't give you more specific advices but I hope you'll find a quick solution by tuning the thresholds. I was considering implementing knn, but I didn't found time for it. If you do, I would be interested in your results.

Concerning the MTCNN I adopted [this](https://github.com/PKUZHOU/MTCNN_FaceDetection_TensorRT) implementation which is based on [this](https://github.com/AlphaQi/MTCNN-light) implementation. For details you should refer those repos. I unfortunately can't give you much information about that.

**Comment by sivak95** [comment]
Hello Daniel,
I was able to get a knn classification with a threshold of 0.5 working which replicated the accuracy I got earlier without compromising on the performance. I need to do some more debugging, but I'll provide an accuracy comparison update in a day or two.

Also, I wanted to mention some changes I made to the cmake file:
1. I was able to comment out line#9 and get it to compile, and I'm not quite sure of the purpose of that line
2. line#14 is not recommended by Davis King as per [this example](https://github.com/davisking/dlib/blob/master/examples/CMakeLists.txt) and the recommended way is to to set the path to dlib source directory using add_subdirectory(). Infact I was getting issues with line#14 despite dlib being detected, so I removed it and used the add_subdirectory() approach
3. Boost is not mentioned in the list of dependencies, but since it was present in the cmake file, I installed and compiled
